// @gatling.io/cli/target/dependencies/index.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.installRecorder = exports.installGatlingJs = exports.versions = void 0;
const promises_1 = __importDefault(require("fs/promises"));
const coursier_1 = require("./coursier");
const graalVm_1 = require("./graalVm");
var versions_1 = require("./versions");
Object.defineProperty(exports, "versions", { enumerable: true, get: function () { return versions_1.versions; } });
const installGatlingJs = async (options) => {
    const downloadDir = `${options.gatlingHome}/tmp/download`;
    await promises_1.default.mkdir(downloadDir, { recursive: true });
    const graalvmHomePath = await (0, graalVm_1.installGraalVm)(options.gatlingHome, downloadDir);
    const coursierPath = await (0, coursier_1.installCoursier)(options.gatlingHome, downloadDir);
    const classpath = await (0, coursier_1.resolveGatlingJsDependencies)(coursierPath, graalvmHomePath, options.postman);
    return {
        graalvmHome: graalvmHomePath,
        coursierBinary: coursierPath,
        jvmClasspath: classpath.trim()
    };
};
exports.installGatlingJs = installGatlingJs;
const installRecorder = async (options) => {
    const downloadDir = `${options.gatlingHome}/tmp/download`;
    await promises_1.default.mkdir(downloadDir, { recursive: true });
    const graalvmHomePath = await (0, graalVm_1.installGraalVm)(options.gatlingHome, downloadDir);
    const coursierPath = await (0, coursier_1.installCoursier)(options.gatlingHome, downloadDir);
    const classpath = await (0, coursier_1.resolveRecorderDependencies)(coursierPath, graalvmHomePath);
    return {
        graalvmHome: graalvmHomePath,
        coursierBinary: coursierPath,
        jvmClasspath: classpath.trim()
    };
};
exports.installRecorder = installRecorder;
